x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/session/SingletonExecutable.java

```diff
@@ -42,12 +44,10 @@ public void execute(Session session, ActionListener<Page> listener) {
 
     @Override
     public String toString() {
-        StringBuilder sb = new StringBuilder();
+        ArrayList<String> outputsWithValue = new ArrayList<>(values.length);
```

Comment:
The initial implementation seems better - using a StringBuilder instead of a list.
The extra "[]" can be added to the builder.

Reply:
What's the concern here? The initial implementation does not separate the entries by comma and `NodeUtils.limitedToString` already provides all the needed functionality.